### PR TITLE
front: electrical profiles: display electrical profiles in gev

### LINKED ISSUE #2919: Display electrical profiles in GEV
### Description and goal

Electrical profiles have to be shown on GEV like horizontal bars using color values from existing catenaries voltage colors (purple for 25000V, red for 1500V, cyan for 15000V, green for other voltages.

### Acceptance criteria

- [x] The display of electrical profiles & modes is composed of horizontal bars, with color from traction mode (or black #333333 for diesel mode)
- [x] The main/high profile has vivid/source color, next steps (more restricted) have same color but become lighter and lighter. The filling of the rectangle must correspond to the mode `used`.
- [x] The value of mode `seen` is displayed inside white rounded (4px radius) inside bar if there's enough space, else it isn't displayed. The text inside always has the main color of voltage mode (doesn't become lighter with profiles step).
- [x] If the value `seen` is differend than the value `used`, the rectangle must be hashed. The rules above still apply (color of hashes = color of mode `used`, text inside the box = mode SEEN with the corresponding color).
- [x] In case of 25000V, only the voltage must be displayed (25000V, 22500V & 20000V).
- [x] In case of 1500V, the voltage **and the step** must be dispayed (for example "1500V A1")
- [x] A popup pops when hovering bar giving color, step value & voltage value/mode
- [x] A checkbox is available in config to switch display of electrical profiles. Off by default.
- [x] A legend is available to explain link between colors &  voltages/modes, displayed on top of the component (not a modal) and activated by an additional round button on bottom right of the chart (try to use same look&feel than map's legend)
- [x] If enough time, please use reference colors from SNCF design (and apply same colors in catenaries in map's representation & legend)

image
image
image

### PR DESCRIPTION
**Pull request by Yohh**
close #2919

### CODE REVIEW

File: front/src/reducers/osrdsimulation/types.ts [hunk 1]
```diff
@@ -91,7 +91,14 @@ export interface Regime {
   error: any;
 }
 
+export interface ModesAndProfiles {
+  start: number;
+  stop: number;
+  used_mode: string;
+  used_profile: string;
```
Comment (alexandredamiron) [inline front/src/reducers/osrdsimulation/types.ts:98]: Please stick to CamelCase

Comment (Yohh) [reply front/src/reducers/osrdsimulation/types.ts:98]: this is the data imported from backend

File: front/src/applications/operationalStudies/components/SimulationResults/SpeedSpaceChart/d3Helpers.js [hunk 2]
```diff
@@ -208,7 +209,106 @@ function drawTrain(
...
+        segment.position_end = source.stop;
+        segment.position_middle = (source.start + source.stop) / 2;
+        segment.lastPosition = dataSimulation.modesAndProfiles.slice(-1)[0].stop;
+        segment.height_start = 4;
+        segment.height_end = 24;
+        segment.height_middle = (segment.height_start + segment.height_end) / 2;
+        segment.usedMode = source.used_mode;
+        segment.usedProfile = source.used_profile;
+
+        // prepare colors
```
Comment (alexandredamiron) [inline front/src/applications/operationalStudies/components/SimulationResults/SpeedSpaceChart/d3Helpers.js]: Please set these consts elsewhere outside the context

File: front/src/applications/operationalStudies/components/SimulationResults/SpeedSpaceChart/ElectricalProfilesLegend.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+interface Profile {
+  mode: string;
+  color: string[];
+  isStriped: boolean;
+}
+
+export const ElectricalProfilesLegend = ({ isActive, setIsActive }: Props) => {
+  const { t } = useTranslation('simulation');
+
+  const legend: Profile[] = [
```
Comment (alexandredamiron) [inline front/src/applications/operationalStudies/components/SimulationResults/SpeedSpaceChart/ElectricalProfilesLegend.tsx]: Please set these consts outside the context of this functionnal prop

File: front/src/applications/operationalStudies/components/SimulationResults/SpeedSpaceChart/d3Helpers.js (hunk 2 above)
Comment (Yohh) [reply front/src/applications/operationalStudies/components/SimulationResults/SpeedSpaceChart/d3Helpers.js]: done

File: front/src/applications/operationalStudies/components/SimulationResults/SpeedSpaceChart/ElectricalProfilesLegend.tsx (hunk 3 above)
Comment (Yohh) [reply front/src/applications/operationalStudies/components/SimulationResults/SpeedSpaceChart/ElectricalProfilesLegend.tsx]: done

**Review by alexandredamiron — APPROVED** [review]

**Pull request merged**